test/FastTests/Client/RavenQueryTests.cs

```diff
@@ -0,0 +1,166 @@
+
+                        Assert.Equal(employee1.FirstName + " " + employee1.LastName, returnedEmployee21.FullName);
+                        Assert.Equal(employee1.Id, returnedEmployee21.EmployeeId);
+                        Assert.Equal(employee2.FirstName + " " + employee2.LastName, returnedEmployee22.FullName);
+                        Assert.Equal(employee2.Id, returnedEmployee22.EmployeeId);
+
+
+                        //Test 3:
+                        var queryNamedClass = session.Query<Employees_ByFirstName.IndexEntry, Employees_ByFirstName>()
+                            .Select(a => new EmployeeProjection
+                            {
+                                FullName = a.FirstName + " " + a.LastName, EmployeeId = Raven.Client.Documents.Queries.RavenQuery.Id(a)
+                            });
+
+                        var queryNamedClassString = queryNamedClass.ToString();
+                        var returnedEmployees3 = queryNamedClass.ToList();
+                        var returnedEmployee31 = returnedEmployees3.First();
+                        var returnedEmployee32 = returnedEmployees3.Last();
+
+                        Assert.Equal(
+                            "from index 'Employees/ByFirstName' as a select { FullName : " +
+                            "a.FirstName+\" \"+a.LastName, EmployeeId : id(a) }",
+                            queryNamedClassString);
+                        Assert.Equal(2, returnedEmployees3.Count);
+
+                        Assert.Equal(employee1.Id, returnedEmployee31.EmployeeId);
+                        Assert.Equal(employee1.FirstName + " " + employee1.LastName, returnedEmployee31.FullName);
+                        Assert.Equal(employee2.Id, returnedEmployee32.EmployeeId);
+                        Assert.Equal(employee2.FirstName + " " + employee2.LastName, returnedEmployee32.FullName);
+
+
+                        //Test 4:
+                        var queryNamedClassWithNestedClass = session.Query<NewEmployee>()
+                            .Select(a => new EmployeeProjection
+                            {
+                                EmployeeId = Raven.Client.Documents.Queries.RavenQuery.Id(a),
+                                FullName = a.FirstName + " " + a.LastName,
+                                Address = new AddressInfo { AddressId = Raven.Client.Documents.Queries.RavenQuery.Id(a), }
+                            });
+
+                        var queryNamedClassWithNestedClassString = queryNamedClassWithNestedClass.ToString();
+                        var returnedEmployees4 = queryNamedClassWithNestedClass.ToList();
+                        var returnedEmployee41 = returnedEmployees4.First();
+                        var returnedEmployee42 = returnedEmployees4.Last();
+
+                        Assert.Equal(
+                            "from 'NewEmployees' as a select { EmployeeId : id(a), FullName : " +
```

Comment:
can we put it in 1 line without concatenation?

Reply:
Done.